Ignore containers and updates functionality, uniformed al `console` to `logger`, HA 2024.11+ Update Entity Fix

SiriosDev opened the pull request:
I've added the ability to exclude individual containers via the boolean label `mqdockerup.ignore` (`true` to ignore), I've also added a warning after the `image not found error` in `DockerhubAdapter.ts` to suggest the use of the label in case of local generated image.

There are no new tests, since I acted directly on the method that handles the “collection” of container inspects. Although now that I think about it from the perspective of using it for just monitoring health or other statistics it might be better to move the filtering to just the checker, or maybe create two different label one for update, one for "all".

And I've converted all the `console.log/error/warn` to `logger.info/error/warn` and fixed the `logger` _format chain_ that previously creating strange indentation.


> Although now that I think about it from the perspective of using it for just monitoring health or other statistics it might be better to move the filtering to just the checker, or maybe create two different label one for update, one for "all".

### Let me know whether to add that👆 in a new commit before merge

MichelFR commented:
First of all thanks for you contribution this will solve some issues people where having with specific images that crashed MqDockerUp. - #185

I think the way you have implemented makes alot of sense. Just setting a docker container label and checking it in MqDockerUp.

In my opinion we should also be able to support ignoring from MqDockerUp side.

Maybe a new setting/.env variable
ignoredContainers=container1,container2 and the seperating it by "," and checking.

or something like that.

MichelFR commented on package.json:
```diff
@@ -24,6 +24,7 @@
     "figlet": "^1.8.0",
     "fs": "^0.0.1-security",
     "lodash": "^4.17.21",
+    "mqdockerup": "file:",
```
Do we need to self reference in the package.json here?

SiriosDev replied:
I don't know, it added itself automatically when I did npm install, maybe it is the explcita version

SiriosDev replied:
removed

SiriosDev commented:
> In my opinion we should also be able to support ignoring from MqDockerUp side.
 Maybe a new setting/.env variable ignoredContainers=container1,container2 and the seperating it by "," and checking.

It makes sense, I can do it today, it shouldn't be difficult


While I'm at it, I'll also add a way to only ignore updates? To allow users to still see stats?

SiriosDev commented:
Im working on request, not merge (i've found an erorr that wil be fixed in next commits)

SiriosDev commented:
Uuuuf 😤technically I've done!🥳 everything I added in this PR should work, plus the new HA 2024.11 changed the way of handling update entities, which required partial rewriting on that🥲, but even that should be working as before (ergo the progress remained in wip I just added a new configuration with an if for the new or legacy method)

MichelFR commented:
Wow thanks for all those commits :)

What exactly has changed in the latest Homeassistant Version?
From what i see only the reporting of an ongoing update has changed?

Previously MqDockerUp was not sending the progress or the in_progress at all. (was not working)

MichelFR commented:
Also another question.
By merging this witht the option to ignore specific containers/container-update-checks we dont need: #385 386

SiriosDev commented:
sorry for the late response, Today was a 🙂🔫 day.

> What exactly has changed in the latest Home assistant Version?

I couldn't find a proper changelog but initially I did try and error then opened this [docs](https://www.home-assistant.io/integrations/update.mqtt/) and removed everything that was removed then renamed what changed and fixed what was no longer compatible.

for example `release_note` (now renamed in `release_summary`), `release_url`, `entity_picture` and `progress` (now `in_progress `+ `update_percentage`) no longer accept `null`
```ts
release_notes: null,
release_url: null,
entity_picture: null,
progress: 0,
```
so I put for now respectively empty string, the url of this repo and the default picutre (at least until I find/find a way to get them from the image itself)
```ts
release_summary: "",
release_url: "https://github.com/MichelFR/MqDockerUp",
entity_picture: "https://raw.githubusercontent.com/MichelFR/MqDockerUp/refs/heads/main/assets/logo_200x200.png",
in_progress: false,
update_percentage: null,
```
as well as all of the `update` field and all of its contents is no more use
```ts
update: {
            state: currentDigest && newDigest && currentDigest !== newDigest ? "available" : "idle",
            installed_version: `${tag}: ${currentDigest?.substring(0, 12)}`,
            latest_version: newDigest ? `${tag}: ${newDigest?.substring(0, 12)}` : null,
            last_check: new Date().toISOString(),
            progress: 0,
            remaining: 0,
          }
```
all of this is now between line 379-464 for the update message and line 323-373 for the progress that even if it doesn't work
> Previously MqDockerUp was not sending the progress or the in_progress at all. (was not working)


I added what there was to add for 2024.11 in case someone works on it in the future

SiriosDev commented:
> Also another question. By merging this witht the option to ignore specific containers/container-update-checks we dont need: #386

It depends (from what I understand) if the user just wants to ignore updates yes, this pr is enough, if on the other hand he wants two different "timers" for `container check` and `update check` then no, they are two different uses (there are conflicting files between the two PRs, if possible also for a matter of lines of code to be adjusted, I would prefer to be merged after #386 so I can adjust mine if that's okay for you)

MichelFR commented:
Alright thanks for clarification seems fine to me then 👍🏽 

after rebase of this PR we a ready to merge, gonna create a new release afterwards

SiriosDev commented:
As soon as I can I'll work on it

SiriosDev commented:
I`ve noticed some little problems following the incorporation of the new changes, unfortunately it`s getting late, tomorrow morning (or for better say this morning in CET 😂) I`ll finish so we can finally merge

SiriosDev commented:
Ok solved, I have: 
* corrected conflicting functionality with the other PR. 
* added a configuration for the log level. 
* added an auto parser for environment varaibles, so I can solve #346 by transforming the obtained strings to a suitable type or leaving them as is
* renamed some configurations for better consistency (⚠️ho also modified the interval ones that now show a deprecation warning in the logs, then removed them permanently sooner or later)

SiriosDev commented:
Ok solved, I've:

- corrected conflicting functionality with the other PR.
- added a configuration for the log level.
- added an auto parser for environment variables, possible solve #346  by transforming the obtained strings to a suitable type or leaving them as is (where needed)
- renamed some configurations for better naming consistency (⚠️also modified the intervals ones that now show a deprecation warning in the logs, then removed them permanently sooner or later)

SiriosDev commented:
If you want to write a release note a moment more understandable I'll leave here quick links to comments with lists of things done:
- https://github.com/MichelFR/MqDockerUp/pull/385#issuecomment-2480758327
- https://github.com/MichelFR/MqDockerUp/pull/385#issuecomment-2463569399

I'd gladly do it for you (or anyway I'd write you something a moment more structured), but for the next 10 days I'll be totally busy, also I make it known to you that unfortunately I didn't find a way, in time, to delete from mqtt (and consequently HA) the ignored items. At the moment they simply don't refresh (or don't re/create)

Maybe when I get free I will open another PR or in the meantime someone else will figure out how to do it

I have tested on my setup, and everything seems to be going, if in case any problems show up once it comes into contact with the public I apologize in advance and if it is necessary I will use some slightly freer evenings to correct

MichelFR commented:
@SiriosDev i would like to merge this now, seems that we cannot rebase this PR to be able to merge right now.

![image](https://github.com/user-attachments/assets/967eff74-1824-4c42-85db-250374dfcfb5)
